scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py

```diff
@@ -132,19 +131,25 @@ def _run_focus(input_workspace,
             logger.warning(f"Skipping focus of run {input_workspace.name()} because it has invalid proton charge.")
             return False
         input_workspace /= vanadium_integration_ws
+        # replace nans created in sensitivity correction
         ReplaceSpecialValues(InputWorkspace=input_workspace, OutputWorkspace=input_workspace, NaNValue=0,
                              InfinityValue=0)
         ApplyDiffCal(InstrumentWorkspace=input_workspace, CalibrationWorkspace=full_calib)
-        ws_d = ConvertUnits(InputWorkspace=input_workspace, Target='dSpacing')
-        focused_sample = DiffractionFocussing(InputWorkspace=ws_d, **df_kwarg)
-        curves_rebinned = RebinToWorkspace(WorkspaceToRebin=vanadium_curves_ws, WorkspaceToMatch=focused_sample)
-        normalised = Divide(LHSWorkspace=focused_sample, RHSWorkspace=curves_rebinned,
-                            AllowDifferentNumberSpectra=True)
-        ApplyDiffCal(InstrumentWorkspace=normalised, CalibrationWorkspace=region_calib)
+        ConvertUnits(InputWorkspace=input_workspace, OutputWorkspace=input_workspace, Target='dSpacing')
+        # rename workspace prior to focussing to avoid errors later
         dspacing_output_name = tof_output_name + "_dSpacing"
-        RenameWorkspace(InputWorkspace=normalised, OutputWorkspace=dspacing_output_name)
-        ConvertUnits(InputWorkspace=normalised, OutputWorkspace=tof_output_name, Target='TOF')
-        DeleteWorkspaces([curves_rebinned, focused_sample, ws_d])
+        RenameWorkspace(input_workspace, dspacing_output_name)
+        # focus over specified region of interest
+        DiffractionFocussing(InputWorkspace=input_workspace, OutputWorkspace=dspacing_output_name,
```

Comment:
I think this call to `DiffractionFocusing` overwrites the whole instrument data - as discussed offline you clone the instrument before this function so the overwriting is OK, but it is a bit confusing and ideally we want to avoid cloning the whole instrument - to this end can we put the sensitivity and unit conversion stuff outside the loop over banks?